### PR TITLE
Fix counter capacity spam

### PR DESCRIPTION
**Pull request by danielnorberg**
# Hey, I just made a Pull Request!

## Description
* propagate CounterCapacityException 
* k8s: handle CounterCapacityException on pod events
* scheduler: handle CounterCapacityException

## Checklist for PR author(s)

- [ ] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java [hunk 1]
```diff
@@ -648,6 +649,8 @@ private void emitPodEvents(Pod pod, RunState runState) {
       } catch (StateTransitionConflictException e) {
         LOG.debug("State transition conflict on kubernetes pod event: {}", event, e);
         return;
+      } catch (CounterCapacityException e) {
+        LOG.debug("Counter capacity exhausted when processing pod event: {}", event, e);
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java]: Don't we need an early return here as well?

File: styx-scheduler-service/src/main/java/com/spotify/styx/state/PersistentStateManager.java [hunk 2]
```diff
@@ -294,34 +290,21 @@ private void updateResourceCounters(StorageTransaction tx, Event event,
...
-        .filter(x -> x._2.getCause() instanceof CounterCapacityException)
-        .map(x -> x._1)
-        // Sort resource ids to get deterministic message contents
-        .sorted()
-        .collect(toList());
 
-    if (!depletedResourceIds.isEmpty()) {
-      MessageUtil.emitResourceLimitReachedMessage(this, runState, depletedResourceIds);
+    var depletedResourceIds = new ArrayList<String>();
+    for (final String resourceId : resourceIds) {
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/state/PersistentStateManager.java]: var

File: styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java (hunk 1 above)
Comment (honnix) [reply styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java]: NVM after checking the complete code.

Comment (honnix) [reply styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java]: NVM after checking the complete code.

Comment (danielnorberg) [reply styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java]: Probably should. Overall this event generation code smells.

File: styx-scheduler-service/src/main/java/com/spotify/styx/state/PersistentStateManager.java (hunk 2 above)
Comment (danielnorberg) [reply styx-scheduler-service/src/main/java/com/spotify/styx/state/PersistentStateManager.java]: 👍

**Comment by honnix** [comment]
Now you will need to cover those new added catches. :D

**Comment by danielnorberg** [comment]
@honnix done!

**Comment by honnix** [comment]
`var` everything? ;)

**Pull request merged**